code/_onclick/hud/interactive_emotes.dm

```diff
@@ -0,0 +1,283 @@
+	var/atom/movable/screen/interaction/interaction
+	if(can_interact(target))
+		var/target_zone = src.zone_selected
+		switch(target_zone)
+			if(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM)
+				if(isxeno(target) && isxeno(src))	//Benos don't high five each other, they slap tails!
+					interaction = /atom/movable/screen/interaction/fist_bump
+				else
+					interaction = /atom/movable/screen/interaction
+			if(BODY_ZONE_PRECISE_L_HAND, BODY_ZONE_PRECISE_R_HAND)
+				interaction = /atom/movable/screen/interaction/fist_bump
+			if(BODY_ZONE_HEAD)
+				interaction = /atom/movable/screen/interaction/headbutt
+
+	if(!interaction)
+		return FALSE
+
+	if(length(target.queued_interactions))
+		for(var/atom/movable/screen/interaction/element AS in target.queued_interactions)
+			if(element.initiator == src)
+				to_chat(src, span_warning("Slow your roll! You're already trying to interact with [target]!"))
+				return FALSE
+
+	interaction = new interaction()
+	interaction.owner = target
+	interaction.initiator = src
+	interaction.register_movement_signals()
+	target.queued_interactions += interaction
+
+	if(target.client && target.hud_used)
+		target.hud_used.update_interactive_emotes()
+
+	interaction.transform = matrix(32, 6, MATRIX_TRANSLATE)
+	animate(interaction, transform = matrix(), time = 2.5, easing = CUBIC_EASING)
+
+	interaction.timer_id = addtimer(CALLBACK(interaction, TYPE_PROC_REF(/atom/movable/screen/interaction, end_interaction), FALSE), interaction.timeout, TIMER_STOPPABLE|TIMER_UNIQUE)
+
+	return TRUE
+
+//Mob interactions
+/atom/movable/screen/interaction
+	name = "high five"
+	desc = "You gonna leave them hanging?"
+	icon = 'icons/mob/screen_alert.dmi'
+	icon_state = "drunk2"	//It looks jolly
+	///Sound filed played when interaction is successful
+	var/interaction_sound = 'sound/effects/snap.ogg'
+	///Who this offer for interaction is being made to
+	var/mob/owner
+	///The mob that initiated the interaction
+	var/mob/initiator
+	///Clear itself after a certain amount of time
+	var/timeout = 10 SECONDS
+	///The reference to the existing timer
+	var/timer_id
+
+/atom/movable/screen/interaction/Initialize(mob/user)
+	. = ..()
+	desc += "\nLeft-click to accept interaction. Right-click or SHIFT + left-click to decline."
+
+///Separate proc to register signals; not on Initialize because owner and initiator are not set yet
+/atom/movable/screen/interaction/proc/register_movement_signals()
+	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(interactees_moved))
+	RegisterSignal(initiator, COMSIG_MOVABLE_MOVED, PROC_REF(interactees_moved))
+
+///What to do when either the owner or the initiating mob moves
+/atom/movable/screen/interaction/proc/interactees_moved()
+	if(!owner.Adjacent(initiator))
+		end_interaction(FALSE)
+
+///Functions to end the interaction
+/atom/movable/screen/interaction/proc/end_interaction(success = TRUE)
+	if(!success)
+		owner.visible_message(failure_message())
+	UnregisterSignal(initiator, COMSIG_MOVABLE_MOVED)
```

Comment:
I don't think these need to be unregistered manually if you're deleting src anyway